File: js/romanNumerals.js
ROMAN_TO_ARABIC_MAP = {
  "M": 1000,
  "D": 500,
  "C": 100,
  "L": 50,
  "X": 10,
  "V": 5,
  "I": 1,
}

const toRoman = function(num) {
  let output = "";
  
  // divide input by 1000, no rounding
  for (let letter in ROMAN_TO_ARABIC_MAP) {
    let arabic = ROMAN_TO_ARABIC_MAP[letter]
    times = Math.trunc(num / arabic);
    // append m as many units as there are
    for (let i = 0; i < times; i++) {
      output += letter
      num -= arabic;
    }
      
  }
  output = condenseMultiples(output);
  return output;
};

condenseMultiples = function(roman) {
  roman = roman.replace(/VIIII/g, "IX");
  roman = roman.replace(/IIII/g, "IV");
  roman = roman.replace(/DCCCC/g, "CM");
  roman = roman.replace(/XXXX/g, "XL");
  roman = roman.replace(/MMMM/g, "ĪV") // I couldn't find the ascii for a 'V' with macron.
  
  return roman;
}

module.exports = { toRoman }

File: python/roman_numerals.py
ROMAN_TO_ARABIC_MAP = {
    "M": 1000,
    "CM": 900,
    "D": 500,
    "CD": 400,
    "C": 100,
    "L": 50,
    "XL": 40,
    "X": 10,
    "IX": 9,
    "V": 5,
    "IV": 4,
    "I": 1,
}


def to_roman(num):
    output = ""
    # iterate our list of possible subs
    for letter, arabic in ROMAN_TO_ARABIC_MAP.items():
      for _ in range(num // arabic):
        # sub, and decrement the input number
        output += letter
        num -= arabic
    return output
